### ISSUE TITLE
[Deal Making Issue] Retrieval stuck at ClientEventWaitForLastBlocks

### ISSUE THREAD

**Issue by dkkapur** [filecoin-project/lotus#6717]
> Note: For security-related bugs/issues, please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).

Please provide all the information requested here to help us troubleshoot "deal making failed" issues.
If the information requested is missing, we will probably have to just ask you to provide it anyway,
before we can help debug.

**Basic Information**
I am the client attempting to retrieve content stored via a deal that was made with offline data transfer.

**Describe the problem**
Retrieval seemed to be working right up until the end where it froze after "ClientEventComplete; ClientEventWaitForLastBlocks".

**Version**
lotus version 1.11.0-rc1+mainnet+git.6bbc5be1b

**Setup**
Using the api.chain.love node through lotus lite

**To Reproduce**
Attempt a retrieval on data CID "bafykbzaced7j2wzyo6tndbfvwmgaer4lx776xzb4dlgxfyu4si4kxl2kkyc3g" from miner "f066596": `lotus client retrieve --miner f066596 bafykbzaced7j2wzyo6tndbfvwmgaer4lx776xzb4dlgxfyu4si4kxl2kkyc3g retrieved.zip`

**Deal status**
`DealStatusWaitingForLastBlocks`

**Lotus daemon and miner logs**

[transferlog.txt](https://github.com/filecoin-project/lotus/files/6789082/transferlog.txt)

<img width="734" alt="retrieval-failure-1" src="https://user-images.githubusercontent.com/2343218/125030918-b4c0e000-e040-11eb-9a97-52419d6a2fd3.png">

** Code modifications **
None

**Comment by dirkmc** [comment]
Looking at the logs it seems like there are connectivity issues between the client and server:
This means that the transfer was restarted because of a disconnection:
```
2021-07-08T20:45:03.885-0700	INFO	dt-chanmon	channelmonitor/channelmonitor.go:426	12D3KooWNfWPDVGCtESxdwsTbXSdtJGzXrtKG8FmTSCaKM3kvLWu-12D3KooWH3mBSF2RFQJFZdqzkVBqSdspoAAunHkViTf5AQhubv1c-1625704128173407001: restart completed successfully
```

"stream reset" means the connection went down
```
2021-07-08T20:52:58.316-0700	WARN	dt-impl	impl/events.go:231	channel 12D3KooWNfWPDVGCtESxdwsTbXSdtJGzXrtKG8FmTSCaKM3kvLWu-12D3KooWH3mBSF2RFQJFZdqzkVBqSdspoAAunHkViTf5AQhubv1c-1625704128173407001 had transport receive error: stream reset
2021-07-08T20:52:58.316-0700	ERROR	dt_graphsync	graphsync/graphsync.go:870	failed to fire transport receive error stream reset: normal shutdown of state machine
2021-07-08T20:52:58.316-0700	WARN	dt-impl	impl/events.go:231	channel 12D3KooWNfWPDVGCtESxdwsTbXSdtJGzXrtKG8FmTSCaKM3kvLWu-12D3KooWH3mBSF2RFQJFZdqzkVBqSdspoAAunHkViTf5AQhubv1c-1625704128173407001 had transport receive error: stream reset
2021-07-08T20:52:58.316-0700	ERROR	dt_graphsync	graphsync/graphsync.go:870	failed to fire transport receive error stream reset: normal shutdown of state machine
2021-07-08T20:52:58.316-0700	WARN	dt-impl	impl/events.go:231	channel 12D3KooWNfWPDVGCtESxdwsTbXSdtJGzXrtKG8FmTSCaKM3kvLWu-12D3KooWH3mBSF2RFQJFZdqzkVBqSdspoAAunHkViTf5AQhubv1c-1625704128173407001 had transport receive error: stream reset
2021-07-08T20:52:58.316-0700	ERROR	dt_graphsync	graphsync/graphsync.go:870	failed to fire transport receive error stream reset: normal shutdown of state machine
```

This one indicates there's an issue connecting to cloudflare
```
2021-07-08T20:55:13.136-0700	INFO	drand	log/log.go:109		{"level": "info", "optimizing_client": "endpoint down when speed tested", "client": "HTTP(\"https://drand.cloudflare.com/\").(+verifier)", "err": "doing request: Get \"https://drand.cloudflare.com/public/1\": context deadline exceeded"}
```

**Comment by dkkapur** [comment]
@dirkmc - its possible that my internet connection reset during this time, I didn't do anything specific but did walk around with my device which may have caused an switch in access point? retrievals are typically resilient / work post an attempted restart though right, which means there could be something on the miner end or with cloudflare in between?

**Comment by dirkmc** [comment]
Yes retrievals should be resilient against restarts 👍 
The issue with the connection to cloudflare appears to be on the client side.
It may also be that there's an issue on the miner end.

**Comment by TippyFlitsUK** [comment]
Hi 👋 

The Legacy Lotus Markets sub-system [reached EOL](https://github.com/filecoin-project/lotus/releases/tag/v1.18.0) at the end of the 31st January 2023.

This ticket is being marked as `won't fix` and closed as the Lotus Team will no longer be making any further fixes or enhancements to the legacy markets subsystem.

Please feel free to re-open this ticket in the new Boost markets sub-system repository at https://github.com/filecoin-project/boost if you feel that it is still relevant.

Many thanks 🙏
